### PR TITLE
Orm relationships

### PR DESCRIPTION
**Pull request by chrisbay**
Still need to give this a once-over edit

### CODE REVIEW

File: src/chapters/orm-relationships/types-of-relationships.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
+Types of Relationships
+======================
+
+Just database tables :ref:`can relate <one-to-many-sql>` to each other, so can classes and objects. In fact, ORM translates relationships between objects into relationships between database rows.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/types-of-relationships.rst]: just as

File: src/chapters/orm-relationships/types-of-relationships.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+This chapter introduces the tools needed to create meaningful relationships using ORM. Let's consider the different types of relationships at a conceptual level. In later sections, we will learn how to implement these relationships using EntityFrameworkCore.
+
+For the examples below, we use four classes:
+
+- ``Event`` - A class representing a coding event.
+- ``EventCategory`` - A class representing categories of coding events.
+- ``EventDetails`` - A class that encapsulates details about a single event, such as description, contact email, location, and so on.
+- ``Tag`` - A piece of metadata labeling an event. You can think of these as topics that an event might include, such as C#, Spring, or JavaScript. An event can cover many topics, so it can have many tags.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/types-of-relationships.rst]: minor but Spring as a topic will mean virtually nothing to a c#-er when they start reading this page. could we add a little note for more context - Spring (a web dev framework for java)?

File: src/chapters/orm-relationships/types-of-relationships.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+.. admonition:: Question
+
+   True/False: Suppose two C# classes, A and B, are in a one-to-many relationship. Then class A must 
+   contain a property for instances of B and B must have a property for instances of A.
+
+   #. True
+   #. False
+
+.. ans: False, A one-to-many relationship may be present without B containing a property A.
+
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/types-of-relationships.rst:140]: good CCs!

File: src/chapters/orm-relationships/lambda-expressions.rst [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+.. index:: ! lambda expression
+
+A **lambda expression** is an *inline* function defined using the symbol ``=>``. This symbol is sometimes called a "fat arrow," and lambda expressions are sometimes called "fat arrow functions." The most basic syntax for creating a lambda expression is:
+
+::
+
+   argument => return expression
+
+The left-hand side of the expression consists of the arguments to be passed into the function, and the right-hand side states the return expression. This value of this expression will be the return value of the function. There are more complicated ways to write a lambda expression, but we will only use this one.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/lambda-expressions.rst]: im not convinced that this sentence is adding anything here:

"This value of this expression will be the return value of the function."

Is it a different idea than what has already been expressed in the previous sentence?

File: src/chapters/orm-relationships/lambda-expressions.rst [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+Lambda expressions can seem confusing until you see them in action. We will
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/lambda-expressions.rst]: We will ...

I think we'll want to say, we'll look at some below ...

File: src/chapters/orm-relationships/lambda-expressions.rst [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      2 4 6 8
+
+Let's look at line 7 in more detail.
+
+.. sourcecode:: csharp
+   :lineno-start: 7
+
+   var doubledNums = nums.Select(x => 2*x);
+
+The ``Select`` method takes a lambda expression as a parameter. The lambda expression is ``x => 2*x``. This is a function that takes a single element of the ``nums`` array (as the parameter ``x``) and returns the result of multiplying it by 2. We can think of the execution of ``Select`` as follows:
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/lambda-expressions.rst]: This is a function that takes a single element of the ``nums`` array, as represented by ``x`` on the left side of the expression, and returns the results of multiplying each of those array elements by 2.

^^ That sentence might push the Hemingway evaluator over the edge but im trying to bridge the earlier explanation of the expression with the detailed example here

File: src/chapters/orm-relationships/lambda-expressions.rst [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+      2 4
+
+In this example, the flow of execution of ``Where`` on line 7 is similar to that of ``Select`` above. The main difference is that instead of *transforming* every element of ``nums``, it is filtered based on the lambda expression ``x => (x % 2 == 0)``.
+
+.. admonition:: Note
+
+   While the examples we have given use arrays, the ``Select`` and ``Where`` methods work with most collection types. More generally, you'll find various specialized methods that use lambdas whenever you encounter a specialized collection type (such as ``DbSet``, as we'll soon see).
+
+Check Your Understanding
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/lambda-expressions.rst:99]: these are good. can we also provide multiple choices though?

File: src/chapters/orm-relationships/one-to-many.rst [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+The first relationship we implement will be between the ``Event`` and ``EventCategory`` classes. We will allow multiple events to be in the same category, but each event will only have one category. Thus, this will be a one-to-many relationship. In this case, we will set up both sides of the relationship, so a many-to-one relationship will result as well.
+
+Setting Up the Relationship - Video
+-----------------------------------
+
+We are now ready to create a relationship between ``Event`` and ``EventCategory``.
+
+.. admonition:: Note
+
+   The starter code for this video is found at the `data-store-controller branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/data-store-controller>`_ of ``CodingEventsDemo``. The final code presented in this video is found on the `one-to-many branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/one-to-many>`_. As always, code along to the videos on your own ``CodingEvents`` project.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst]: persistent-controller branch

File: src/chapters/orm-relationships/one-to-many.rst [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+In the ``Event`` class, replace the ``Type`` property with a new property of type ``EventCategory``. In order for EntityFrameworkCore to be able to persist relationships between ``Event`` objects and ``EventCategory`` objects, we must also have a property that stores the ``Id`` of the given ``EventCategory``.
+
+.. sourcecode:: csharp
+   :lineno-start: 14
+
+   public EventCategory Category { get; set; }
+
+   public int CategoryId { get; set; }
+
+The ``CategoryId`` property functions as a `foreign key <foreign-key>`_. EF will create a ``CategoryId`` column in the ``Event`` table. The value of this column for a given row will determine which row in the ``Category`` table is related to the given event. Our code is now set up so that each ``Event`` will knows about its ``EventCategory`` object, and that relationship persists.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst]: we need :ref: in front of this foreign key reference

File: src/chapters/orm-relationships/one-to-many.rst [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+               Value = category.Id.ToString(),
+               Text = category.Name
+            });
+      }
+   }
+
+The ``Value`` of each ``SelectListItem`` will be the ``Id`` of the given category. The ``Id`` of a category is unique (in fact, it functions as a primary key) while the ``Name`` may not be. Therefore, we must use ``Id`` for the ``value`` attribute.
+
+Since we no longer have a no-arg constructor, we must add one.
+
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst:76]: have we established elsewhere that ef requires a no arg constructor?

File: src/chapters/orm-relationships/one-to-many.rst [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+Now, let's remove all references to ``EventType`` in the project.
+
+Open ``AddEventViewModel``, which is in the ``ViewModels`` directory. Recall that this ViewModel represents the data that is needed to display and process the form used to create new ``Event`` instances. Replace its ``Type`` and ``EventType`` properties with similar properties that use ``EventCategory``.
+
+.. sourcecode:: csharp
+   :lineno-start: 22
+
+   [Required(ErrorMessage = "Category is required")]
+   [Display(Name = "Category")]
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst]: provably not crucial but we do not introduce this attribute in the validation section.

and if you do keep it here, can we remove the label text in the view sourcecode below?

File: src/chapters/orm-relationships/one-to-many.rst [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+   public AddEventViewModel()
+   {
+   }
+
+There is one more reference to ``EventType``, and it is in ``Views/Events/Add.cshtml``. Within that file, update the ``select`` input and its label to reference our new ``Category`` and ``Categories`` properties.
+
+.. sourcecode:: html
+   :lineno-start: 21
+
+   <label asp-for="CategoryId">Category</label>
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst:89]: to the comment above, if we keep the display name attribute, do we need the inner label text here?

File: src/chapters/orm-relationships/one-to-many.rst [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+Since we no longer have a no-arg constructor, we must add one.
+
+.. sourcecode:: csharp
+   :lineno-start: 42
+
+   public AddEventViewModel()
+   {
+   }
+
+There is one more reference to ``EventType``, and it is in ``Views/Events/Add.cshtml``. Within that file, update the ``select`` input and its label to reference our new ``Category`` and ``Categories`` properties.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst]: there is another*

File: src/chapters/orm-relationships/one-to-many.rst [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+.. admonition:: Note
+
+   The starter code for this video is found at the `one-to-many branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/one-to-many>`_ of ``CodingEventsDemo``. The final code presented in this video is found on the `refactoring-controller branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/refactoring-controller>`_. As always, code along to the videos on your own ``CodingEvents`` project.
+
+.. todo: Add refactoring-controller video
+
+Refactoring the Controller and View - Text
+------------------------------------------
+
+Our ``EventsController`` requires a few updates to continue to work as it has been.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/one-to-many.rst]: how about : Our ``EventsController`` requires a few updates now that Event objects reference EventCategory objects?

just some more-specific language around this refactor

File: src/chapters/orm-relationships/many-to-many.rst [hunk 15]
```diff
@@ -0,0 +1,690 @@
...
+====================================
+
+This section explores how we can work with many-to-many relationships in ASP.NET using EntityFrameworkCore. To do so, we need a class that we can relate to ``Event`` in a many-to-many fashion. This is the ``Tag`` class.
+
+The ``Tag`` Model Class
+-----------------------
+
+Between the last section and this one, we have added a new persistent model class to the code, ``Tag``. This class represents a tag of the type used for blog or social media posts. For us, a tag will be a topical label that can be applied to any event. Thus, we'll eventually relate ``Event`` and ``Tag`` to each other in a many-to-many way.
+
+The steps to add this code follow the exact same process that we used to add a `persistent EventCategory class <orm1-exercises>`_, so we won't go through them in detail here.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: this also needs :ref:

File: src/chapters/orm-relationships/many-to-many.rst [hunk 16]
```diff
@@ -0,0 +1,690 @@
...
+      .Include(e => e.Category)
+      .Single(e => e.Id == id);
+
+This method takes a boolean lambda expression and filters the ``Context.Events`` collection down to the *one* event that satisfies ``e.Id == id``. In other words, it finds the single event with ``Id`` matching the path parameter.
+
+.. admonition:: Note
+
+   We use ``Single`` instead of ``Find`` here because we also need to call ``Include`` to eagerly fetch the ``Category`` property. ``Include`` can not be chained with ``Find``.
+
+Finally, we're ready for the view, which is straightforward. Here are the contents of ``Views/Events/Detail.cshtml``:
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: the last change is also described as straightforward so i'd drop it from here

File: src/chapters/orm-relationships/many-to-many.rst [hunk 17]
```diff
@@ -0,0 +1,690 @@
...
+
+Join Tables
+-----------
+
+To relate data in a many-to-many fashion in a relational database requires a new type of SQL table.
+
+
+One-to-many relationships are established at the database level by the use of a foreign key column on one side of the relationship. Our ``Event`` table has a foreign key column: ``CategoryId``. 
+
+For a given row in ``Event``, the column ``CategoryId`` contains the primary key of the row in ``event_category`` that the ``Event`` row is related to.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: can this be broken down into two smaller statements?

File: src/chapters/orm-relationships/many-to-many.rst [hunk 18]
```diff
@@ -0,0 +1,690 @@
...
+-----------
+
+To relate data in a many-to-many fashion in a relational database requires a new type of SQL table.
+
+
+One-to-many relationships are established at the database level by the use of a foreign key column on one side of the relationship. Our ``Event`` table has a foreign key column: ``CategoryId``. 
+
+For a given row in ``Event``, the column ``CategoryId`` contains the primary key of the row in ``event_category`` that the ``Event`` row is related to. 
+
+The only difference is the *number* of different ``Event`` rows that may have the same value of ``CategoryId``. The ``event``/``event_category`` relationship is many-to-one, so *many* event rows may have the same ``CategoryId`` value.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: the only difference from what? its not clear to me so far that a comparison is being or has been established

File: src/chapters/orm-relationships/many-to-many.rst [hunk 19]
```diff
@@ -0,0 +1,690 @@
...
+
+One-to-many relationships are established at the database level by the use of a foreign key column on one side of the relationship. Our ``Event`` table has a foreign key column: ``CategoryId``. 
+
+For a given row in ``Event``, the column ``CategoryId`` contains the primary key of the row in ``event_category`` that the ``Event`` row is related to. 
+
+The only difference is the *number* of different ``Event`` rows that may have the same value of ``CategoryId``. The ``event``/``event_category`` relationship is many-to-one, so *many* event rows may have the same ``CategoryId`` value. 
+
+.. index:: ! join table
+
+Using foreign and primary keys to create many-to-many relationships is a bit trickier. In order to relate rows in ``Event`` to rows in ``Tag`` we need need a third table, known as a **join table**. A join table consists of two columns, each of which is a foreign key column to another table. Each row in a join table represents a relationship between one row in each of the two tables. This technique enables many-to-many relationships.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: In order to relate rows in ``Event`` to rows in ``Tag``, .. (add a comma)

Each row in a join table represents a relationship between one row in each of the two tables. - can we say "in each of the two tables being joined"?

File: src/chapters/orm-relationships/many-to-many.rst [hunk 20]
```diff
@@ -0,0 +1,690 @@
...
+   * - Id
+     - Name
+   * - 4
+     - ios
+   * - 5
+     - spring
+   * - 6
+     - java
+
+A join table for these two tables would be called ``EventTag``, and would have two columns, ``EventId`` and ``TagId``. Each of these columns are foreign key columns into their respective tables.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: is the join table EventTag or EventTags?

File: src/chapters/orm-relationships/many-to-many.rst [hunk 21]
```diff
@@ -0,0 +1,690 @@
...
+The ``EventTag`` Model - Video
+------------------------------
+
+Let's create a new model class, ``EventTag``, to model a join table for ``Event`` and ``Tag`` classes. 
+
+.. admonition:: Note
+
+   If you want to verify what code this video starts with, check out the `event-detail-view <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-detail-view>`_ branch. If you want to verify what code this video ends with, check out the `event-tag-model <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-tag-model>`_ branch.
+
+   The starter code for this video is found at the `event-detail-view branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-detail-view>`_ of ``CodingEventsDemo``. The final code presented in this video is found on the `event-tag-model branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-tag-model>`_. As always, code along to the videos on your own ``CodingEvents`` project.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: i think you can do with just one of these notes

File: src/chapters/orm-relationships/many-to-many.rst [hunk 22]
```diff
@@ -0,0 +1,690 @@
...
+
+   protected override void OnModelCreating(ModelBuilder modelBuilder)
+   {
+      modelBuilder.Entity<EventTag>()
+            .HasKey(et => new { et.EventId, et.TagId });
+   }
+
+The method ``OnModelCreating`` can be overridden from the base class, ``DbContext``, in order to provide additional configuration for the data store. In this case, we add code that configures ``EntityTag`` to have a composite primary key consisting of the properties/columns ``EventId`` and ``TagId``.
+
+This completes configuration of our join class. Before proceeding, create and apply a database migration. After doing so, verify that there is a new join table, ``EventTag``.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: im not sold that readers will be at the point of creating their own migration and applying it. applying it maybe, but id just give them more advice to turn to a previous migration command to copy and name it something descriptive

File: src/chapters/orm-relationships/many-to-many.rst [hunk 23]
```diff
@@ -0,0 +1,690 @@
...
+
+Adding a ``Tag`` to an ``Event`` - Video
+----------------------------------------
+
+Now that we have established a many-to-many relationship between ``Event`` and ``Tag`` we can write controller and view code to allow users to add tags to events.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: Now that we have established a many-to-many relationship between ``Event`` and ``Tag``,

File: src/chapters/orm-relationships/many-to-many.rst [hunk 24]
```diff
@@ -0,0 +1,690 @@
...
+
+         public AddEventTagViewModel()
+         {            
+         }
+      }
+   }
+
+This class models the data that is need to render and process our form. In order to add a tag to an event, our ``POST`` handler will need to know the IDs of the two objects in question. Therefore, our ViewModel has required ``EventId`` and ``TagId`` properties. It also contains an ``Event`` property, which we will use to display details (such as the event name) in the view.
+
+Finally, the ViewModel has a property ``List<SelectListItem> Tags``. As with previous forms containing a dropdown, this property will be used to populate the ``select`` element containing the all tag options.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: containing all of the tag options.

File: src/chapters/orm-relationships/many-to-many.rst [hunk 25]
```diff
@@ -0,0 +1,690 @@
...
+.. sourcecode:: csharp
+   :lineno-start: 59
+
+   [HttpPost]
+   public IActionResult AddEvent(AddEventTagViewModel viewModel)
+   {
+      if (ModelState.IsValid)
+      {
+
+            int eventId = viewModel.EventId;
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: trivial again but i think the indent in this block is one-too-many.

see what i did there 😉

File: src/chapters/orm-relationships/many-to-many.rst [hunk 26]
```diff
@@ -0,0 +1,690 @@
...
+            context.EventTags.Add(eventTag);
+            context.SaveChanges();
+
+            return Redirect("/Events/Detail/" + eventId);
+      }
+
+      return View(viewModel);
+   }
+
+This handler takes in a ``AddEventTagViewModel`` object which will be created via model binding. Assuming validation passes (that is, both ``EventId`` and ``TagId`` are not ``null``) we create a new ``EventTag`` object and save it to the database. Then we redirect to the detail view for the given event.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: This action method* takes in an* ``AddEventTagViewModel`` ...

File: src/chapters/orm-relationships/many-to-many.rst [hunk 27]
```diff
@@ -0,0 +1,690 @@
...
+
+#. **Line 93** - Filters the ``EventTags`` set to include only objects related to the given ``Event``.
+#. **Line 94** - Forces eager loading of the ``Tag`` property of those ``EventTag`` objects.
+#. **Line 95** - Converts the ``DbSet`` to a list.
+
+.. admonition:: Note
+
+   You might be wondering why we have to query ``context.EventTags``. Indeed, it would be convenient of we could just reference a ``Tags`` property from the ``Event`` class. But notice that there *is no such property* in ``Event``. The many-to-many relationship is defined by the data in ``EventTag``, so we must use this class in order to access related objects.
+
+Now let's move into ``EventDetailsViewModel``. Here, we can add data related to an event's tags, so that it can be passed into the view.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: Here, we add data related to an event's tags to pass into the view.

File: src/chapters/orm-relationships/many-to-many.rst [hunk 28]
```diff
@@ -0,0 +1,690 @@
...
+
+First, add a new string property named ``TagText``.
+
+.. sourcecode:: csharp
+   :lineno-start: 14
+
+   public string TagText { get; set; }
+
+Then, in the constructor add a parameter to represent the list of all of ``EventTag`` objects associated with a given ``Event``. Use this parameter to set the value of ``TagText``.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: Then in the constructor,

File: src/chapters/orm-relationships/many-to-many.rst [hunk 29]
```diff
@@ -0,0 +1,690 @@
...
+      {
+            TagText += ("#" + eventTags[i].Tag.Name);
+            if (i < eventTags.Count - 1)
+            {
+               TagText += ", ";
+            }
+      }
+   }
+
+We build up the contents of ``TagText`` by looping over ``eventTags`` and appending tag names, separated by commas. For example, if an event has tags with names ``"java"``, ``"csharp", and ``"object-oriented"``, then the ``TagList`` will be ``"#java, #csharp, #object-oriented"``.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: the syntax highlighting is a little off here but i think you can get away with removing it completely from ``"java"``, ``"csharp", and ``"object-oriented"``

File: src/chapters/orm-relationships/many-to-many.rst [hunk 30]
```diff
@@ -0,0 +1,690 @@
...
+               context.SaveChanges();
+            }
+
+            return Redirect("/Events/Detail/" + eventId);
+      }
+
+      return View(viewModel);
+   }
+
+Lines 68-71 query for existing ``EventTag`` objects that have the some ``EventId``/``TagId`` pair. In other words, ``existingItems`` will be empty unless that given event already has the given tag. Before creating and saving a new ``EventTag`` object, we check the size of ``existingItems``, skipping this step if a the event already has the tag.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: if the event already has the tag.

File: src/chapters/orm-relationships/many-to-many.rst [hunk 31]
```diff
@@ -0,0 +1,690 @@
...
+
+Display Items With a Given Tag - Video
+--------------------------------------
+
+In addition to seeing which tags are on an event, we would also like to see all events with a specific tag.
+
+.. admonition:: Note
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst:466]: again, we can drop one of these items in the note

File: src/chapters/orm-relationships/many-to-many.rst [hunk 32]
```diff
@@ -0,0 +1,690 @@
...
+The view is similar to other listings that we have created.
+
+.. sourcecode:: html
+   :linenos:
+
+   @model List<CodingEventsDemo.Models.EventTag>
+
+   @if (Model.Count == 0)
+   {
+      <h1>No elemnts with the given tag</h1>
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: elements

File: src/chapters/orm-relationships/many-to-many.rst [hunk 33]
```diff
@@ -0,0 +1,690 @@
...
+
+Finally, we add links to the name of each tag in our tag index.
+
+``Views/Tag/Index.cshtml``:
+
+.. sourcecode:: html
+   :lineno-start: 18
+
+   <tr>
+         <td>@tag.Id</td>
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst:532]: again i think the indent is too far here

File: src/chapters/orm-relationships/many-to-many.rst [hunk 34]
```diff
@@ -0,0 +1,690 @@
...
+         <td><a asp-controller="Tag" asp-action="Detail" asp-route-id="@tag.Id">@tag.Name</a></td>
+   </tr>
+
+Start the app up and test. Viewing the main tag listing should allow you to click on each tag name and view the events that have that tag.
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/many-to-many.rst]: and test user behavior

File: src/chapters/orm-relationships/exercises.rst [hunk 35]
```diff
@@ -0,0 +1,27 @@
+Exercises: The Early Bird Gets the ORM!
+=======================================
+
+Now that we have gotten very familiar with our ``CodingEvents`` application, let's design some additional features.
+As you work on your ``CodingEvents`` application, you may have been inspired by `Meetup <https://www.meetup.com/>`_!
```
Comment (CarlyLanglois) [inline src/chapters/orm-relationships/exercises.rst]: can we remove either the "!" from the title or this statement -too many too close diminishes their worth

Comment (CarlyLanglois) [reply src/chapters/orm-relationships/exercises.rst]: ok actually there a 5 ! on this short page. id argue we can remove them all

**Review by CarlyLanglois — APPROVED** [review]
seconding that the many to many page is huge. Id break it up where we start talking about join tables/models

File: src/chapters/orm-relationships/one-to-many.rst (hunk 10 above)
Comment (chrisbaylewis) [reply src/chapters/orm-relationships/one-to-many.rst:76]: The no-arg constructor is for model binding. I added that clarification.

**Comment by chrisbay** [comment]
@CarlyLanglois Thanks for the detailed review! I made all suggested changes. I also broke up the many-to-many content up into two pages. Am going to go ahead and merge, since you marked this "approved."

**Pull request merged**